.github/workflows/justtest.yaml

```diff
@@ -0,0 +1,699 @@
+          # debug and release builds?
+          #
+          # First, there are tests that uses error injections,
+          # which are only available on the debug build.
+          #
+          # Second, the debug build has assertions enabled, so it
+          # can spot a problem that passes unseen on the release
+          # build.
+          #
+          # Third, the debug build has Lua's strict mode enabled
+          # by default.
+          #
+          # OTOH, the release build is what a customer typically
+          # uses, so it is the primary target. It feels
+          # counter-intuitive to verify the primary target with
+          # less tests that the additional targets.
+          #
+          # However, a release deployment job anyway performs the
+          # release build and it is different from this one in
+          # fact (at least the dependencies are linked statically
+          # and a specific building container is used). So, maybe,
+          # it is better to concentrate on developer needs in this
+          # workflow and reduce testing on the release build?
+          - debug
+          - release
+          # The idea is to verify that tarantool compiles with
+          # clang. At first, it is a code health check. And next,
+          # some developers use it.
+          #
+          # clang is known for its good analysis and warnings
+          # regarding potentially problematic code, but some of
+          # these nice linter capacities need analysis passes that
+          # are run on -O2, but not -O0. This way we can only get
+          # these warnings on RelWithDebInfo, but not on Debug.
+          #
+          # Seems like there is no reason to additionally verify
+          # the Debug build on clang.
+          #
+          # TODO: Whether we have to run all the tests with the
+          # clang build?
+          - clang
+
+        # Define cmake-build-type.
+        include:
+          - build-conf: debug
+            cmake-build-type: Debug
+          - build-conf: release
+            cmake-build-type: RelWithDebInfo
+          - build-conf: clang
+            cmake-build-type: RelWithDebInfo
+
+    # }}} Build configurations
+
+    steps:
+      # {{{ Install build dependencies
+
+      # Disable man-db triggers. They're executed after each
+      # $(apt-get install <...>) call and it sometimes takes
+      # about a minute.
+      #
+      # https://github.com/actions/runner-images/issues/10977
+      # https://stackoverflow.com/a/77841777/1598057
+      - name: Disable man-db triggers
+        run: |
+          echo "set man-db/auto-update false" | sudo debconf-communicate
+          sudo dpkg-reconfigure man-db
+
+      - name: Install build dependencies
+        run: sudo apt-get install libreadline-dev
+
+      - name: Setup clang
+        if: ${{ matrix.build-conf == 'clang' }}
+        run: |
+          wget https://apt.llvm.org/llvm.sh
```

Comment:
Probably it's worth to add retries, see https://github.com/tarantool/tarantool/blob/c86a5e14df23c07bd379490600e4b47618fc292f/.github/workflows/perf_micro.yml#L143-L150